### DISCUSSION TITLE
How to add unexpected one-time bills

### DISCUSSION THREAD

**Discussion by obizbu** [firefly-iii/firefly-iii#5233] [Firefly III]
If I have an unexpected bill, like an auto repair or hospital bill, that is sent to my house, but I haven't paid yet - how do I represent that?
Bills don't make sense, since it seems Bills are for things that repeat at some frequency (e.g. monthly water bill)
Transactions require both source and destination accounts, so they don't make sense. (Unless, I guess I could create an accounts payable account?)
I looked at Liabilities, which seems it might work, but I may get multiple bills for the same account (e.g. for auto repair: a bill for inspection, then 2 weeks later a bill for repair, then another 2 weeks later another bill for delivery) and it would be nice to represent those. Perhaps I could use transactions, but again, I don't understand what the source, destination accounts would be. 

Can you help me out?

**Comment by JC5** [comment]
You create a transaction, from a source (asset account) to a new destination account.

I don't even know what an accounts payable account is.

https://docs.firefly-iii.org/firefly-iii/concepts/accounts/

**Comment by obizbu** [comment]
Hmm. I'm still confused. 
So let me simplify:
I have a medical bill. I haven't paid it yet. 
Obviously, I'd add an expense account for the clinic. And then I'd create an expense transaction...
But that's where I get confused - which account is source, and which is destination?
I wouldn't want to have the account "source" be my asset account or checking account, because I haven't paid anything yet. So what do I use?

I'm kind of thinking my use case is for a bill that is non-recurring. Do you think that's a valid use case?

**Reply by wazabees** [reply]
> But that's where I get confused - which account is source, and which is destination?

Which ever account you are paying the bill from is the source (Asset account), and the destination is the clinic (Expense account). Expense accounts are the same as payee, i.e. the receiver of your money.

Set the date in the date picker to the date the bill is to be paid/due date.

**Comment by obizbu** [comment]
I'm starting to see it. Appreciate your help. 
One challenge I still have is that if I follow what you laid out above, once the due date has passed, firefly will assume the expense has been paid. To put it another way, I don't have a way to capture overdue expenses. Am I thinking about this incorrectly?

**Reply by wazabees** [reply]
> I'm starting to see it. Appreciate your help. One challenge I still have is that if I follow what you laid out above, once the due date has passed, firefly will assume the expense has been paid. To put it another way, I don't have a way to capture overdue expenses. Am I thinking about this incorrectly?

I'm a new Firefly III user as well, so take my replies with a grain of salt. As far as I can see in Firefly, there is no way to "clear" a transaction once the money has actually left your account. It is however possible to [reconcile your accounts](https://docs.firefly-iii.org/firefly-iii/advanced-concepts/reconcile/), which would catch any transactions that have not happened.

**Reply by obizbu** [reply]
I think I've figured out how to make it work the way I want it to...see the following ledger:

Pre-requisites (all starting balances = 0)
- Employer Account (where I get paid from)
- Checking account (Asset)
- Credit Card Account (Asset)
- Hospital Account (Expense)
- Hospital "Bill" or Liability (Liability)


**Day 0**
- Get Paid
- - Income Transaction. 100 FROM "employer account" to "checking account"

**Day 1**
- Get a vaccine for 75 at the hospital
- No action

**Day 2**
- Receive bill from hospital
- - Expense Transaction. 75 FROM "Hospital Liability" to "Hospital Account"

**Day 5**
- Pay hospital bill
- - Expense Transaction. 75 FROM "Credit Card" to "Hospital Liability"

**Day 10**
- Pay Credit Card bill
- - Transfer Transaction. 75 FROM "Checking Account" to "Credit Card"

Net Ending Balances:
- Employer Account (where I get paid from) = -100
- Checking account (Asset) = 25
- Credit Card Account (Asset) = 0
- Hospital Account (Expense) = 75
- Hospital "Bill" or Liability (Liability) = 0
This ending state should represent that I am in the clear with the hospital (I don't have any outstanding amount due)

Is this right??

**Reply by wazabees** [reply]
I suppose you can do it like that. However, I don't understand the "Hospital liability/bill".
On Day 2, when you receive the bill, just enter it as a transaction FROM credit card TO Hospital and set the date to Day 5.

**Reply by JC5** [reply]
It's really over the top. You're moving around 75 between many accounts for no reason at all. But if it works for you.

**Reply by obizbu** [reply]
Thanks both. I guess what I'm trying to capture is the fact that on day 2 I have an outstanding balance. If I directly credit my hospital account on day 2, then it appears as if I have no outstanding balance. The hospital liability gives me a place to put that outstanding debt.

~~This brings me back to an earlier point. Could we add an option to the "Bills" where the repeat frequency is "none"? That would allow me to enter this in as a bill, rather than having to have separate hospital account and hospital liability. Alternatively, is there a way to set an end date to a bill? I would imagine that would be a need regardless (e.g. paying electrical bill monthly, then moving and having electricity covered in the rent)... If that was possible, it would be possible for me to create a recurring bill that ends after 1 time, and I can capture my hospital bill above that way...~~

EDIT: Never mind. I didn't realize you can't have a transaction that _pays_ a bill. I don't have another way to accomplish what I'm proposing above...

Thanks again for the discussion/insight. I'm really excited about using this tool

**Reply by diamant-x** [reply]
This is an interesting use case, there's a thin line between liabilities and bills from a philosophical point of view.

In FF3 the closest is Bills represent -recurring- invoices to come (but not necessarily valid yet). For example, electricity bills are expected each month, but I could cancel the contract and then the bill would be no more in the future. This means Bills are only an orientation, and mostly useful for short-term invoices only with certain degree of recurrence.
If you have outstanding invoices with longer due dates, then I agree the best way is through liabilities. End of the day, (from your example) the healthcare provides is giving you a free-of-interest credit amount to be paid later in the future for a medical expense you have already incurred.

End of the day I would say, If you need it to impact your current net-worth: go for the liabilities (more complex to keep track), if you don't need it to reflect in your net-worth and just need to remember that is coming: create a new bill for it (you can set it to start and end date on same expected invoice-payment date to stop recurrence).

**Comment by tmllull** [comment]
Hi! I'm looking for this kind of bills, and I want to explain my case (just in case).

I'm doing running races around my country, at the end of this month I will have an "extra" expense (hotel), and I know the exact amount because I already booked (but not paid). This kind of expenses only occurs one time, but I think has the entity to be a bill (and not just an extra expense).

For me, would be a very useful to see in the "Bills to pay" section, in the main page, this expected bills (has not paid yet, but I know will be), because I can adjust my budget in this "special" month. I'm thinking a workaround for this, and maybe I can create a bill with any repetition time, and disable when has been paid.

I don't know if there are other option to represent these expected outcomes that occur only one time.

Thanks for this great app, and this active community!

**Reply by JC5** [reply]
Thanks!

In the scope of existing bills, adding a one-time entry isn't hard. The use case to track one time unique bills isn't something Firefly III supports at the moment. At best, you could fill a piggy bank but that's not perfect either. I'll think about it!

**Comment by cachho** [comment]
+1 for adding one-time bills.

Let's say I want to get a new roof and I know that it will be 10000 upfront and 20000 half way and another 20000 when done. I want to create a new bill that says 50000 and link it in each expense. This is a simple example, but this way I can always see what I have yet to expect to pay.

**Comment by jochemvangrondelle** [comment]
@JC5 would it be a pragmatic solution to reuse subscriptions to solve for one off future bills, so that a given subscription has a one off schedule?
